common/meta/sled-store/src/sled_tree.rs

```diff
@@ -467,12 +479,72 @@ impl SledTree {
     }
 }
 
+pub struct TransactionSledTree<'a> {
+    pub txn_tree: &'a TransactionalTree,
+}
+
+impl TransactionSledTree<'_> {
+    pub fn key_space<KV: SledKeySpace>(&self) -> AsTxnKeySpace<KV> {
+        AsTxnKeySpace::<KV> {
+            inner: self,
+            phantom: PhantomData,
+        }
+    }
+
+    // fn get<KV: SledKeySpace>(&self, key: &KV::K) -> common_exception::Result<Option<KV::V>> {
+    //     let got = self
+    //         .txn_tree
+    //         .get(KV::serialize_key(key)?)
+    //         .map_err_to_code(ErrorCode::MetaStoreDamaged, || format!("get: {}", key))?;
+    //     let v = match got {
+    //         None => None,
+    //         Some(v) => Some(KV::deserialize_value(v)?),
+    //     };
+    //     Ok(v)
+    // }
+
+    fn insert<KV>(
+        &self,
+        key: &KV::K,
+        value: &KV::V,
+    ) -> Result<Option<KV::V>, SledUnabortableTransactionError>
+    where
+        KV: SledKeySpace,
+    {
+        let k = KV::serialize_key(key).unwrap();
+        let v = KV::serialize_value(value).unwrap();
+
+        let prev = self
+            .txn_tree
+            .insert(k, v)
+            .map_err(SledUnabortableTransactionError::from)?;
```

Comment:
Why do we need to map the error here?

The caller should still be in a sled txn context, and the sled::transaction expects a `ConflictableTransactionResult` error right?